### PR TITLE
COMMERCE-11003 We need to filter for allowedCPDefinitionOptionValueRels always

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-11003
Resent from https://github.com/liferay-commerce/liferay-portal/pull/3577

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 1]
```diff
@@ -890,6 +890,80 @@ definition {
 		}
 	}
 
+	@description = "This is a test case for COMMERCE-11003. Assert only the product options that are added to SKUs can be displayed in the product detail widget"
+	@priority = 5
+	test CanDisplayOptionsOnlyAddedToProductSKUs {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:895]:
```suggestion
	test CanAssertProductOptionSelectorDisplaysOnlySkuOptions {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 2]
```diff
@@ -890,6 +890,80 @@ definition {
...
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("When creating a simple product with four values in an option and enabling SKU Contributor for that option") {
+			CommerceProducts.openProductsAdmin();
+
+			CPCommerceCatalog.newProductsEntry(
+				productCatalog = "Minium",
+				productName = "Test",
+				productType = "Simple");
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:908]: The creation of a product can be done using the macro `CommerceJSONProductsAPI._addCommerceProduct()`. Please check the code to see how it's used.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 3]
```diff
@@ -890,6 +890,80 @@ definition {
...
+	@description = "This is a test case for COMMERCE-11003. Assert only the product options that are added to SKUs can be displayed in the product detail widget"
+	@priority = 5
+	test CanDisplayOptionsOnlyAddedToProductSKUs {
+		property portal.acceptance = "true";
+
+		task ("Given a Minium site") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("When creating a simple product with four values in an option and enabling SKU Contributor for that option") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:902]:
```suggestion
		task ("When a simple product with four values sku contributor option is created") {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 4]
```diff
@@ -890,6 +890,80 @@ definition {
...
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/p/test");
+		}
+
+		task ("And opens the dropdown option values selector") {
+			Click(
+				key_optionName = "City",
+				locator1 = "CommerceFrontStore#FS_PRODUCT_OPTIONS_SELECT");
+		}
+
+		task ("And only the 2 values set for the SKUs before are shown") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:952]:
```suggestion
		task ("Then only the 2 values set for the SKUs before are shown") {
```

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 5]
```diff
@@ -890,6 +890,80 @@ definition {
...
+
+		task ("And adding only two of the four values as SKUs from SKUs tab of the Product") {
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceSKUs.addSKU(optionValuesList = "Delhi:1,Mumbai:2");
+		}
+
+		task ("Then admin navigates to the product detail page") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/p/test");
+		}
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:944]: Please do the assertion using a non-admin user(buyer)

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase [hunk 6]
```diff
@@ -890,6 +890,80 @@ definition {
...
+			CPCommerceCatalog.newProductsEntry(
+				productCatalog = "Minium",
+				productName = "Test",
+				productType = "Simple");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+
+			CommerceProductOptions.createProductOption(key_optionName = "City");
+
+			Refresh();
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase:914]: Can you try to find a way not to use the `Refresh()` in this case?

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]
@jaydawu please have a look at the comments! Thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Review by lorenzolapescara — APPROVED** [review]

**Comment by jaydawu** [comment]
ci:forward

**Pull request closed without merging**